modules/services/agentless-scanning/main.tf

```diff
@@ -221,7 +221,7 @@ data "aws_iam_policy_document" "key_policy" {
       identifiers = [
         "arn:aws:iam::${var.agentless_account_id}:root",
         var.trusted_identity,
-        aws_iam_role.agentless[0].arn,
+        (var.deploy_global_resources) ? aws_iam_role.agentless[0].arn : var.main_region_agentless_role_arn,
```

Comment:
Hi @blkt @mbreitung , can you please give more context on why this line L224 needs to change?

Because the role ARN is basically region-less. Hence, for both main/primary region and secondary regions, this will eventually be the same value i.e Agentless IAM role's ARN. Do we really have a use-case to introduce this new variable - `var. main_region_agentless_role_arn`?

Reply:
when applying this in the secondary regions, the iam_role is not created and fails
```
aws_kms_replica_key.agentless_replica[0]: Refreshing state... [id=mrk-808967ed839848d59cfe690df939d4b2]
╷
│ Error: Invalid index
│ 
│   on main.tf line 224, in data "aws_iam_policy_document" "key_policy":
│  224:         aws_iam_role.agentless[0].arn,
│     ├────────────────
│     │ aws_iam_role.agentless is empty tuple
│ 
│ The given key does not identify an element in this collection value: the
│ collection has no elements.
```

Reply:
sounds like a dependency expression is missing then. fixing this with a new input variable seems like a hack, and will also require corresponding upstream work to set it appropriately.  Avoiding the need to do this is desirable.

Reply:
Ah, I see.  I think since we now moved to creating KMS Primary key in each of the regions (for both single and org install), this started happening where each time it would need to refer to the Agentless IAM role's ARN string, which is created just once in main/primary region when `var.deploy_global_resources = true`.